src/engine/strat_engine/backstore/util.rs

```diff
@@ -36,3 +39,44 @@ pub fn hw_lookup(dev_node_search: &Path) -> EngineResult<Option<String>> {
 
     result
 }
+
+// Probes a block device using the libblkid library and returns Some(None) if the device is empty or
+// Some(String) with the String having similar content as the output of `blkid -p <device>.
+pub fn device_signature(path: &Path) -> Result<Option<String>, blkid::BlkidError> {
+
+    let d = blkid::BlkId::new(path)?;
+
+    d.enable_partitions()?
+        .enable_superblocks()?
+        .set_superblock_flags(blkid_sys::BLKID_SUBLKS_USAGE | blkid_sys::BLKID_SUBLKS_TYPE |
+                              blkid_sys::BLKID_SUBLKS_BADCSUM |
+                              blkid_sys::BLKID_SUBLKS_UUID)?;
+
+    // There is potential for the probe to fail if the device is in use/locked, lets loop a bit
+    // to ride out these potentially temporary errors.
+    let mut num_retries = 5;
+    let probe_result = loop {
+        match d.do_safe_probe() {
+            Ok(p) => break p,
+            Err(e) => {
+                num_retries -= 1;
+                if num_retries == 0 {
+                    return Err(e);
+                }
+                thread::sleep(time::Duration::from_millis(100));
+            }
+        }
+    };
+
+    Ok(match probe_result {
+           1 => None,
+           -2 => Some(String::from("Ambiguous device usage detected")),
+           _ => {
```

Comment:
What the what? This is a pretty strange API we're dealing with. 1? -2? everything else? Since we're already engaged with the upstream, can we make this a little clearer, maybe with a nice enum?

Reply:
If you peek at tmy PR I ask this specific question: https://github.com/cholcombe973/blkid/pull/3  as I was thinking enum too.  However, it was merged as is.  I'll approach upstream with another PR which makes this a bit nicer.